src/main/java/io/cryostat/credentials/Credentials.java

```diff
@@ -41,38 +42,58 @@
 import org.jboss.resteasy.reactive.RestResponse.ResponseBuilder;
 import org.projectnessie.cel.tools.ScriptException;
 
-@Path("/api/v2.2/credentials")
+@Path("/api/v4/credentials")
 public class Credentials {
 
     @Inject TargetMatcher targetMatcher;
     @Inject Logger logger;
 
     @GET
     @RolesAllowed("read")
-    public V2Response list() {
-        List<Credential> credentials = Credential.listAll();
-        return V2Response.json(
-                Response.Status.OK,
-                credentials.stream()
-                        .map(
-                                c -> {
-                                    try {
-                                        return Credentials.safeResult(c, targetMatcher);
-                                    } catch (ScriptException e) {
-                                        logger.warn(e);
-                                        return null;
-                                    }
-                                })
-                        .filter(Objects::nonNull)
-                        .toList());
+    public Response list() {
+        try {
+            List<Credential> credentials = Credential.listAll();
+            List<Map<String, Object>> results =
+                    credentials.stream()
+                            .map(
+                                    c -> {
+                                        try {
+                                            return Credentials.safeResult(c, targetMatcher);
+                                        } catch (ScriptException e) {
+                                            logger.warn(e);
+                                            return null;
+                                        }
+                                    })
+                            .filter(Objects::nonNull)
+                            .toList();
+
+            JsonObject response = new JsonObject();
+            response.put(
+                    "meta", new JsonObject().put("type", "application/json").put("status", "OK"));
```

Comment:
This is the same thing as as V2Response, just manually encoded instead of using a utility structure. I think this endpoint should just `return Response.ok(results).type(MediaType.APPLICATION_JSON).build();` for the new API format.